File: app/src/main/java/com/ninpou/qbits/MainActivity.java
package com.ninpou.qbits;

import android.os.Bundle;
import android.text.Spannable;
import android.text.SpannableString;
import android.text.style.TypefaceSpan;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.ninpou.qbits.capture.CaptureFragment;
import com.ninpou.qbits.request.RequestFragment;
import com.ninpou.qbits.tool.ToolFragment;

import java.util.Objects;

public class MainActivity extends BaseActivity {
    private final Fragment[] fragments = new Fragment[]{
            CaptureFragment.newInstance(),
            RequestFragment.newInstance(),
            ToolFragment.newInstance()
    };
    private int lastFragmentIndex = 0;

    private BottomNavigationView.OnNavigationItemSelectedListener onNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_capture:
                    switchFragment(0);
                    return true;
                case R.id.navigation_request:
                    switchFragment(1);
                    return true;
                case R.id.navigation_tool:
                    switchFragment(2);
                    return true;
            }
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        ActionBar actionBar = getSupportActionBar();
        Objects.requireNonNull(actionBar).setTitle("Netrunner");
        Objects.requireNonNull(actionBar).setDisplayShowHomeEnabled(true);
        //actionBar.setIcon(R.mipmap.ic_launcher_new);

    }

    private void initView() {
        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener);
        FragmentTransaction transaction;
        transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.pager, fragments[0]);
        transaction.show(fragments[0]);
        transaction.commit();
    }

    private void switchFragment(int index) {
        if (index == lastFragmentIndex) return;
        FragmentTransaction transaction;
        transaction = getSupportFragmentManager().beginTransaction();
        transaction.hide(fragments[lastFragmentIndex]);
        if (!fragments[index].isAdded()) {
            transaction.add(R.id.pager, fragments[index]);
        }
        transaction.show(fragments[index]);
        lastFragmentIndex = index;
        transaction.commitAllowingStateLoss();
    }

}

File: app/src/main/java/com/ninpou/qbits/MainActivity.kt
package com.ninpou.qbits

import android.os.Bundle
import androidx.fragment.app.FragmentTransaction
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.ninpou.qbits.capture.CaptureFragment
import com.ninpou.qbits.request.RequestFragment

import com.ninpou.qbits.util.APP_ACTIVITY
import com.ninpou.qbits.util.setActionBar
import com.ninpou.qbits.util.switchFragment


class MainActivity : BaseActivity() {
    private val fragments = arrayOf(
            CaptureFragment.newInstance(),
            RequestFragment.newInstance(),
            MainFragment.newInstance()
    )

    private val onNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_capture_ui -> {
                switchFragment(0, fragments)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_request_ui -> {
                switchFragment(1, fragments)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_tool_ui -> {
                switchFragment(2, fragments)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        APP_ACTIVITY = this //getting link of current activity
        initView()
        setActionBar(getString(R.string.app_name))
    }

    private fun initView() {
        val navigation = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        navigation.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)
        val transaction: FragmentTransaction = supportFragmentManager.beginTransaction()
        transaction.setCustomAnimations(R.anim.fade_in_main, R.anim.fade_out_main);
        transaction.replace(R.id.nav_host_fragment, fragments[0])
        transaction.show(fragments[0])
        transaction.commit()
    }
}